src/fieldml_util_routines.f90

```diff
@@ -202,23 +202,11 @@ SUBROUTINE FIELDML_IO_FINALISE( FIELDML_INFO, ERR, ERROR, * )
 
     IF(ASSOCIATED(FIELDML_INFO)) THEN
       FML_ERR = Fieldml_Destroy( FIELDML_INFO%FML_HANDLE )
-    
-      FIELDML_INFO%FML_HANDLE = FML_INVALID_HANDLE
-      FIELDML_INFO%NODES_HANDLE = FML_INVALID_HANDLE
-      FIELDML_INFO%NODES_ARGUMENT_HANDLE = FML_INVALID_HANDLE
-      FIELDML_INFO%MESH_HANDLE = FML_INVALID_HANDLE
-      FIELDML_INFO%ELEMENTS_HANDLE = FML_INVALID_HANDLE
-      FIELDML_INFO%ELEMENTS_ARGUMENT_HANDLE = FML_INVALID_HANDLE
-      FIELDML_INFO%XI_HANDLE = FML_INVALID_HANDLE
-      FIELDML_INFO%XI_ARGUMENT_HANDLE = FML_INVALID_HANDLE
-      FIELDML_INFO%NODE_DOFS_HANDLE = FML_INVALID_HANDLE
-!      fieldmlInfo%elementDofsHandle = FML_INVALID_HANDLE
-!      fieldmlInfo%constantDofsHandle = FML_INVALID_HANDLE
-    
       CALL LIST_DESTROY( FIELDML_INFO%COMPONENT_HANDLES, ERR, ERROR, *999 )
       CALL LIST_DESTROY( FIELDML_INFO%BASIS_HANDLES, ERR, ERROR, *999 )
       CALL LIST_DESTROY( FIELDML_INFO%BASIS_CONNECTIVITY_HANDLES, ERR, ERROR, *999 )
       CALL LIST_DESTROY( FIELDML_INFO%BASIS_LAYOUT_HANDLES, ERR, ERROR, *999 )
+      DEALLOCATE(FIELDML_INFO)
     ELSE
       CALL FlagError("FieldML Info is not associated.",ERR,ERROR,*999)
```

Comment:
Sorry, I should have seen this earlier. We don't have else flag error if it is not associated in finalise routines. This is so finalise can be called on error condition from initialise. The (user callable) destroy routines (which are effectively just a wrapper for finalise) do, however, check for association.